python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -46,14 +50,33 @@ def run_subcommand(argv: list[str]) -> Result[str, str]:
         LOGGER.debug("Running edit subcommand")
         res_edit_config = core_utils.parse_args(main_parser, argv[1:], EditServerConfig)
         LOGGER.debug(f"{res_edit_config.is_ok()=}")
-        res_servers = res_edit_config.and_then(_run_editor_servers)
         out: Result[str, str] = result.do(
             #
             Ok(",".join(res_servers_ok))
             #
-            for res_servers_ok in res_servers
+            for edit_config in res_edit_config
+            for cli_config in res_cli_config
+            for res_servers_ok in _run_editor_servers(edit_config, cli_config.config_path)
```

Comment:
What does this do?

Reply:
This is admittedly arcane syntax that unpacks the result of _run_editor_servers (if it's an Ok). You can think of it kind of like a Result comprehension, sort of like a list comprehension that flattens a 2d list.

Does that answer the question? What would you name differently?

Reply:
See do notation in the readme: https://github.com/rustedpy/result

Reply:
Let me take a look and read to understand, I've never seen 3 for loops list comprehension within the same inline (still not super familiar how inline tabs work in Python tbh, so it's just not clear to me what parts are nested or not)

Reply:
Oh thanks, this example makes it super clear to me!